File: app/src/main/java/hibernate/v2/testyourandroid/ui/adapter/MainFragmentPagerAdapter.java
package hibernate.v2.testyourandroid.ui.adapter;

import android.content.Context;
import androidx.fragment.app.Fragment;
import androidx.fragment.app.FragmentManager;
import androidx.fragment.app.FragmentPagerAdapter;
import android.view.View;
import android.widget.TextView;

import hibernate.v2.testyourandroid.R;
import hibernate.v2.testyourandroid.ui.fragment.MainAboutFragment;
import hibernate.v2.testyourandroid.ui.fragment.MainTestFragment;

public class MainFragmentPagerAdapter extends FragmentPagerAdapter {

	private String tabTitles[];

	private Context mContext;

	/**
	 * Constructor of the class
	 */
	public MainFragmentPagerAdapter(FragmentManager fm, Context mContext) {
		super(fm);
		tabTitles = mContext.getResources().getStringArray(R.array.main_tab_title);
		this.mContext = mContext;
	}

	/**
	 * Returns the number of pages
	 */
	@Override
	public int getCount() {
		return 2;
	}

	/**
	 * This method will be invoked when a page is requested to create
	 */
	@Override
	public Fragment getItem(int position) {
		Fragment fragment;
		switch (position) {
			/* Android tab is selected */
			case 0:
				fragment = MainTestFragment.newInstance();
				break;
			case 1:
			default:
				fragment = new MainAboutFragment();
				break;
		}
		return fragment;
	}

	public View getTabView(int position) {
		View v = View.inflate(mContext, R.layout.custom_tab, null);
		TextView tv = v.findViewById(R.id.tabTitleTv);
		tv.setText(tabTitles[position]);
		return v;
	}
}

File: app/src/main/java/hibernate/v2/testyourandroid/ui/adapter/MainFragmentPagerAdapter.kt
package hibernate.v2.testyourandroid.ui.adapter

import android.content.Context
import android.view.View
import android.widget.TextView
import androidx.fragment.app.Fragment
import androidx.fragment.app.FragmentManager
import androidx.fragment.app.FragmentStatePagerAdapter
import hibernate.v2.testyourandroid.R
import hibernate.v2.testyourandroid.ui.fragment.MainAboutFragment
import hibernate.v2.testyourandroid.ui.fragment.MainTestFragment

class MainFragmentPagerAdapter(
        private val context: Context,
        fm: FragmentManager
) : FragmentStatePagerAdapter(fm, BEHAVIOR_RESUME_ONLY_CURRENT_FRAGMENT) {

    private val tabTitles: Array<String> = context.resources.getStringArray(R.array.main_tab_title)
    /**
     * Returns the number of pages
     */
    override fun getCount(): Int = 2

    /**
     * This method will be invoked when a page is requested to create
     */
    override fun getItem(position: Int): Fragment {
        return when (position) {
            0 -> MainTestFragment.newInstance()
            1 -> MainAboutFragment()
            else -> MainAboutFragment()
        }
    }

    fun getTabView(position: Int): View {
        val v = View.inflate(context, R.layout.custom_tab, null)
        val tv = v.findViewById<TextView>(R.id.tabTitleTv)
        tv.text = tabTitles[position]
        return v
    }

}